File: C++/M01_ProgrammingBasics/L04_ForLoop/Exercises/Solutions/P02_HalfSumElement.cpp
#include <iostream>
#include <climits>
#include <algorithm>
using namespace std;

int main() {

	int n;
	cin >> n;

	int maxNum = INT16_MIN;

	int sum = 0;
	for (int i = 0; i < n; i++) {
		int input;
		cin >> input;

		sum += input;

		maxNum = max(maxNum, input);
	}

	sum -= maxNum;


	if (sum == maxNum) {
		cout << "Yes" << endl;
		cout << "Sum = " << maxNum;
	} else {
		cout << "No" << endl;
		cout << "Diff = " << abs(sum - maxNum) << endl;
	}

	return 0;
}

File: Python/M01_ProgrammingBasics/L04_ForLoop/Exercises/Solutions/P02_HalfSumElement.py
import sys

max_num = -sys.maxsize
total_num = 0

n = int(input())

for each in range(1, n + 1):
    number = int(input())

    if max_num < number:
        max_num = number

    total_num += number

total_num = total_num - max_num

if total_num == max_num:
    print("Yes")
    print(f"Sum = {max_num}")
else:
    print("No")
    print(f"Diff = {abs(max_num - total_num)}")
